pywinauto/controls/uia_controls.py

```diff
@@ -1302,6 +1308,111 @@ def check_button(self, button_identifier, make_checked, exact=True):
+        sub_item = None
+        items = menu.items()
+        if items:
+            sub_item = items[idx]
+        self._activate(sub_item, is_last)
+        return sub_item
+
+    # ----------------------------------------------------------------
+    items = buttons
+
+    def item_by_path(self, path, exact=False):
+        """
+        Walk the items in this toolbar to find the item specified by a path
+
+        The path is specified by a list of items separated by '->'. Each item
+        can be either a string (can include spaces) e.g. "Save As" or a zero
+        based index of the item to return prefaced by # e.g. #1.
+
+        These can be mixed as necessary. For example:
+            - "#0->Save As",
+            - "Tools->#0->Configure"
+
+        * **path** - Path to the specified item. **Required**.
+        * **exact** - If false, text matching will use a 'best match' fuzzy algorithm. If true, will try to find the
+                      item with the given name. (Default False). **Optional**
+        """
+        toolbar_items = [p.strip() for p in path.split("->")]
+        items_cnt = len(toolbar_items)
+        if items_cnt == 0:
+            raise IndexError()
+        for item in toolbar_items:
+            if not item:
+                raise IndexError("Empty item name between '->' separators")
+
+        def next_level_menu(parent_menu, item_name, is_last):
+            if item_name.startswith("#"):
+                return self._sub_item_by_idx(parent_menu, int(item_name[1:]), is_last)
+            else:
+                return self._sub_item_by_text(parent_menu, item_name, exact, is_last)
+
+        try:
+            if items_cnt == 1:
+                menu = next_level_menu(self, toolbar_items[0], items_cnt == 1)
+                return menu
+            else:
+                menu = self
+                new_descendants = []
+                for i in range(items_cnt):
+                    descendants_before = self.top_level_parent().descendants()
+                    if len(new_descendants) == 0:
+                        menu = next_level_menu(menu, toolbar_items[i], items_cnt == i + 1)
+                    else:
+                        new_descendants.append(menu)
+                        try:
+                            for ctrl in new_descendants[::-1]:
+                                try:
+                                    menu = next_level_menu(ctrl, toolbar_items[i], items_cnt == i + 1)
+                                except AttributeError:
+                                    pass
+                        except (findwindows.ElementNotFoundError,
+                                findbestmatch.MatchError):
+                            raise IndexError()
+                    descendants_after = self.top_level_parent().descendants()
+                    new_descendants = list(set(descendants_after) - set(descendants_before))
+
+        except AttributeError:
+            raise IndexError()
```

Comment:
Which lines are supposed to fail with such `AttributeError`? Can we make more detailed error message based on this info?

Reply:
I suggest to remove this exception. If `items_count==1`, we will get a detailed error message from `findbestmatch.find_best_match`. We can leave one exception in case when `items_count > 1`. What do you think?